.github/workflows/pull-request.yaml

```diff
@@ -0,0 +1,80 @@
+name: Build and test a pull request
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  ci:
+    name: Run CI build
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    steps:
+      - name: Set up Go 1.14
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.14
+
+      - name: Install github.com/vbatts/git-validation
+        working-directory: /tmp
+        run: go get -u github.com/vbatts/git-validation
+
+      - name: Install github.com/cpuguy83/go-md2man
+        env:
+          VERSION: "v1.0.10"
+        working-directory: /tmp
+        run: go get "github.com/cpuguy83/go-md2man@${VERSION}"
+
+      - name: Check out source code
+        uses: actions/checkout@v2
+
+      - name: Validate DCO
+        env:
+          DCO_VERBOSITY: "-q"
+          GIT_CHECK_EXCLUDE: "./vendor:./script/validate/template"
+        run: git-validation "${DCO_VERBOSITY}" -run DCO,short-subject,dangling-whitespace
+
+      - name: Check go.mod matches the source code in the module
+        run: go mod tidy
+
+      - name: Vendor go modules
+        run: go mod vendor
+
+      - name: Checking for unused packages in vendor
+        run: |
+          git diff --exit-code -- go.sum go.mod vendor/
+          untracked=$(git ls-files --others vendor | wc -l | awk '{ print $1 }')
+          [[ "${untracked}" == "0" ]]
```

Comment:
I think it would be more efficient to handle vendors validation (and update) through a dedicated Dockerfile to be able to reproduce locally. I['ve started to work on it](https://github.com/crazy-max/docker-distribution/commit/3d74672436dc96a00b68a8852c6d53c7142ab6a8#diff-ca6580e530ff7bbb5ce286e2613578e7c815652cac23b866537b30b4a1f98557R1) but didn't saw your PR since yesterday ^^

Reply:
As reference, you can find more info about the migration plan of buildx and buildkit in docker/buildx#451.

Reply:
Gotcha. Yeah, that looks interesting. Need to look into that, thanks.

Btw, if you are keen on getting actions working I'm happy to close this PR. I just wanted to get the ball rolling quickly on this. I'm not hung on getting this into master. As long as it's done by anyone I'm happy.

Reply:
@milosgajdos Yeah sure I can work on it if you don't mind!

Reply:
Not at all! Thanks. I'm just glad someone else other than me cares about this now :-)